torch/_dynamo/source.py

```diff
@@ -291,6 +291,26 @@ def name(self):
         return f"{self.base.name()}.node.{self.prop}()"
 
 
+@dataclasses.dataclass(frozen=True)
+class UnionFindMetadataSource(ChainedSource):
+    prop: str
+
+    def __post_init__(self):
+        assert self.base is not None
+
+    def reconstruct(self, codegen):
+        return [
+            *self.base.reconstruct(codegen),
+            codegen.create_load_attr(self.prop),
+        ]
+
+    def guard_source(self):
+        return self.base.guard_source()
+
+    def name(self):
+        return f"torch.nested._internal.union_find.get_union_find().get_metadata({self.base.name()})['{self.prop}']"
```

Comment:
how does this work? can we be sure that the UF will be in the same state at guard time?

Reply:
Is the concern that if the metadata does not actually exist for the input offsets it would error while computing guards, and not properly recompile? Will double check this.